File: Visual Basic .NET/PersistSettings/PredefinedSettingsItem.vb
'
' Xceed DataGrid for WPF - SAMPLE CODE - Persist User Settings Sample Application
' Copyright (c) 2007-2024 Xceed Software Inc.
' 
' [PredefinedSettingsItem.vb]
'  
' Structure used for the predefined combo box binding.
' 
' This file is part of the Xceed DataGrid for WPF product. The use 
' and distribution of this Sample Code is subject to the terms 
' and conditions referring to "Sample Code" that are specified in 
' the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
'

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel

Namespace Xceed.Wpf.DataGrid.Samples.PersistSettings

  Public Class PredefinedSettingsItem
    Implements INotifyPropertyChanged
#Region "DisplayName Property"

    Public Property DisplayName() As String
      Get
        Return m_displayName
      End Get
      Set(ByVal value As String)
        If value <> m_displayName Then
          m_displayName = value
          Me.OnPropertyChanged(New PropertyChangedEventArgs("DisplayName"))
        End If
      End Set
    End Property

#End Region

#Region "XmlUri Property"

    Public Property XmlUri() As String
      Get
        Return m_xmlUri
      End Get
      Set(ByVal value As String)
        If value <> m_xmlUri Then
          m_xmlUri = value
          Me.OnPropertyChanged(New PropertyChangedEventArgs("XmlUri"))
        End If
      End Set
    End Property

#End Region

#Region "INotifyPropertyChanged Members"

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Protected Overridable Sub OnPropertyChanged(ByVal e As PropertyChangedEventArgs)
      RaiseEvent PropertyChanged(Me, e)
    End Sub

#End Region

#Region "PRIVATE FIELDS"

    Private m_displayName As String
    Private m_xmlUri As String

#End Region
  End Class
End Namespace

File: CSharp/PersistSettings/PredefinedSettingsItem.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Persist User Settings Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [PredefinedSettingsItem.cs]
 *  
 * Structure used for the predefined combo box binding.
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System.ComponentModel;

namespace Xceed.Wpf.DataGrid.Samples.PersistSettings
{
  public class PredefinedSettingsItem : INotifyPropertyChanged
  {
    #region DisplayName Property

    public string DisplayName
    {
      get
      {
        return m_displayName;
      }
      set
      {
        if( value == m_displayName )
          return;

        m_displayName = value;
        this.OnPropertyChanged( new PropertyChangedEventArgs( "DisplayName" ) );
      }
    }

    #endregion DisplayName Property

    #region XmlUri Property

    public string XmlUri
    {
      get
      {
        return m_xmlUri;
      }
      set
      {
        if( value == m_xmlUri )
          return;

        m_xmlUri = value;
        this.OnPropertyChanged( new PropertyChangedEventArgs( "XmlUri" ) );
      }
    }

    #endregion XmlUri Property

    #region INotifyPropertyChanged Members

    public event PropertyChangedEventHandler PropertyChanged;

    protected virtual void OnPropertyChanged( PropertyChangedEventArgs e )
    {
      if( this.PropertyChanged != null )
        this.PropertyChanged( this, e );
    }

    #endregion INotifyPropertyChanged Members

    #region PRIVATE FIELDS

    private string m_displayName;
    private string m_xmlUri;

    #endregion PRIVATE FIELDS
  }
}
